lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_method_settings.rb

```diff
@@ -0,0 +1,81 @@
+require_relative "./helpers"
+
+########################################################################
+# AwsApiGatewayMethodSettings is the +api_gateway_method_settings+ terrform
+# resource.
+#
+# {https://www.terraform.io/docs/providers/aws/r/api_gateway_method_settings.html}
+########################################################################
+class GeoEngineer::Resources::AwsApiGatewayMethodSettings < GeoEngineer::Resource
+  include GeoEngineer::ApiGatewayHelpers
+
+  validate -> {
+    validate_required_attributes([
+                                   :rest_api_id,
+                                   :stage_name,
+                                   :method_path,
+                                   :settings
+                                 ])
+  }
+
+  # Must pass the rest_api as _rest_api resource for additional information
+  after :initialize, -> { self.rest_api_id = _rest_api.to_ref }
```

Comment:
Where does _rest_api attribute come from? Should it be a required attribute?

Reply:
Moreover, the validation says that rest_api_id is required. Is _rest_api the parent rest api resource, and that just works?

Reply:
It's provided manually similar to most of the other API Gateway resources. It should be an `api_gateway_rest_api` object.

Reply:
`rest_api_id` is derived from the `_rest_api` provided by the user. Since this happens before validation, the validation passes. This is, again, quite similar to all of the other API Gateway resources.